plugins/module_utils/ansible_freeipa_module.py

```diff
@@ -566,6 +566,21 @@ def gen_intersection_list(user_list, res_list):
     return list(set(res_list or []).intersection(set(user_list or [])))
 
 
+def concat_attr_list(res, *args):
+    """
+    Get the lists of members to pass on `gen_*` methods as a res_list argument.
+
+    This function should be used to get members (usually users,
+    external users, hosts, external hosts, etc.) with any action and any state.
+
+    It is returning the concatenation of all attributes provided by user.
+    """
+    res_list = []
+    for attribute in args:
+        res_list += res.get(attribute, [])
```

Comment:
This might break when executing in client context, as data returned might be a tuple instead of a list.

What if you create a set with the first arg and add all other args to the set, and return a list created from this set? Would this incur in some performance issue?

Reply:
So... this do work in Python 3.11. There is a small risk that the behavior is fixed in the future, but I have nothing against using it now.

```
Python 3.11.1 (main, Jan  6 2023, 00:00:00) [GCC 12.2.1 20221121 (Red Hat 12.2.1-4)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = []
>>> b = (1, 2, 3)
>>> a += b
>>> a
[1, 2, 3]
>>> a + b
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate list (not "tuple") to list
```

Reply:
I also checked `concat_attr_list` with client context (with tuples) and with the mix of lists and tuples (although it's probably not possible in IPA):

tuples:
```
res = {'externalhost': ('external.host', 'internal.host'), 'memberhost_host': ('master.ipadomain.test', 'client.ipadomain.test')}
>>> args = ('memberhost_host', 'externalhost')
>>> res_list = []
>>> for attribute in args:
...     res_list += res.get(attribute, [])
...     
>>> print(res_list)
['master.ipadomain.test', 'client.ipadomain.test', 'external.host', 'internal.host']
```

Mix:
```
res = {'externalhost': ('external.host', 'internal.host'), 'memberhost_host': ['master.ipadomain.test', 'client.ipadomain.test']}
>>> args = ('memberhost_host', 'externalhost')
>>> res_list = []
>>> for attribute in args:
...     res_list += res.get(attribute, [])
...     
>>> print(res_list)
['master.ipadomain.test', 'client.ipadomain.test', 'external.host', 'internal.host']
```

Reply:
Yes, it will work. As I said, our only risk is python getting its operators fixed.

If you don't tell them, I won't.  ;-)